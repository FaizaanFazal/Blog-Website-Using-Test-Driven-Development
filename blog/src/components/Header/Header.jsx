import "../../styles/Header.scss";
import images from "../../utils/images";
import Navbar from "../Navbar/Navbar";

const Header = () => {
  return (
    <header className="header flex flex-col" style = {{background: `url(${images.hd_base}) center/cover no-repeat`}}>
      <Navbar />
    </header>
  )
}

export default Header