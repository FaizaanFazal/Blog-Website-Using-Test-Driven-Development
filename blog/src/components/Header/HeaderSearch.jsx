import React, { useState } from 'react';

export default function HeaderSearch() {
  const [inputValue, setInputValue] = useState('');

  const handleInput = (event) => {
    setInputValue(event.target.value);
  };

  return (
    <div className="hd-search">
      <form>
        <div className="form-group flex items-stretch bg-white">
          <span className="form-gp-icon items-center justify-center">
            <i className="bi bi-search" />
          </span>
          <input
            type="text"
            value={inputValue}
            className="form-gp-input"
            placeholder="Search article"
            onChange={handleInput}
            data-testid="inputSearch"
          />
          <button type="button" disabled={!inputValue} className="form-gp-btn text-white bg-primary btn">Search</button>
        </div>
      </form>
    </div>
  );
}
