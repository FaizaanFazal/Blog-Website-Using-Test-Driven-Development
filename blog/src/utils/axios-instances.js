// import axios from 'axios';

// export const axiosInstance = axios.create({
//   baseURL: 'http://localhost:8000/',
// });
import axios from 'axios';

const placeholderApi = axios.create({
  baseURL: 'http://localhost:8000',
});

export default placeholderApi;
