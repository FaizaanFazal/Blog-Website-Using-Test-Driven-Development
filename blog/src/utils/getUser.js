export const getUser = () => {
    const userString = localStorage.getItem('user');
    if (userString) {
      const user = JSON.parse(userString);
      return user;
    } else {
      return null;
    }
  };