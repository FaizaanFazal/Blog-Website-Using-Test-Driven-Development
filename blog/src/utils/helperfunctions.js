import { fireEvent } from '@testing-library/react';
import images from './images';

export const isProperImageURL = (url) => {
  /* eslint-disable-next-line no-useless-escape */
  const regex = /^(https?|ftp):\/\/(([a-z\d]([a-z\d-]*[a-z\d])?\.)+[a-z]{2,}|localhost)(\/[-a-z\d%_.~+]*)*(\?[;&a-z\d%_.~+=-]*)?(\#[-a-z\d_]*)?$/i;
  return regex.test(url);
};
export const isProperEmail = (email) => {
  const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
  return emailRegex.test(email);
};
export const isProperPass = (pass) => {
  const passregex = /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[@$!%*#?&])[A-Za-z\d@$!%*#?&]{8,}$/;
  return passregex.test(pass);
};
export function backspace(element) {
  let actuallyTyped = element.value;

  const backspaceKey = {
    key: 'Backspace',
    code: 8,
    inputType: 'deleteContentBackward',
  };

  const sharedEventConfig = {
    key: backspaceKey.key,
    charCode: backspaceKey.code,
    keyCode: backspaceKey.code,
    which: backspaceKey.code,
    modifier: backspaceKey.modifier,
  };
  const downEvent = fireEvent.keyDown(element, sharedEventConfig);

  if (downEvent) {
    actuallyTyped = actuallyTyped.slice(0, -1);

    fireEvent.input(element, {
      target: { value: actuallyTyped },
      inputType: backspaceKey.inputType,
      bubbles: true,
      cancelable: true,
    });
  }

  fireEvent.keyUp(element, sharedEventConfig);
}

export const small = [
  {
    id: 1,
    title: '10 Tips for Budget-Friendly Travel',
    author: 'Elena Rodriguez',
    authorImage: { src: images.author1, alt: 'Elena Rodriguez' },
    date: '15 August',
    image: { src: images.blogImg1, alt: '10 Tips for Budget-Friendly Travel Blog Header Image' },
    verified: true,
    slug: 'post1',
    content: "test Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.",
  },
];
export const greater = [
  {
    id: 1,
    title: '10 Tips for Budget-Friendly Travel',
    author: 'Elena Rodriguez',
    authorImage: { src: images.author1, alt: ' Elena' },
    date: '15 August',
    image: { src: images.blogImg1, alt: '10 Tips for Budget-Friendly Travel Blog Header Image' },
    slug: 'post1',
    verified: true,
    content: "Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n \n Traveling on a budget doesnt have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n \n Traveling on a budget doesnt have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank.Traveling on a budget doesn't have to compromise your experience. Here are 10 tips to make the most of your journey without breaking the bank. \n",
  }];
