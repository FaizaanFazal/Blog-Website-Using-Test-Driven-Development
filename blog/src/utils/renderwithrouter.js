import {render} from '@testing-library/react'
import * as userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom'
import {MemoryRouter} from 'react-router-dom'

export const renderWithRouter = (ui, {route = '/'} = {}) => {
  window.history.pushState({}, 'Test page', route)

  return {
    user: userEvent.setup,
    ...render(ui, {wrapper: MemoryRouter}),
  }
}