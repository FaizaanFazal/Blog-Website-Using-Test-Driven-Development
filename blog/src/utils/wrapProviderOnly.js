import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import reduxStore from '../Redux/store';
import '@testing-library/jest-dom';

export * from '@testing-library/react';

const AllTheProviders = ({ children }) => (
  <Provider store={reduxStore}>
    {children}
  </Provider>
);

export const renderWithProvidersOnly = (ui, options) => render(<AllTheProviders>{ui}</AllTheProviders>, options);

// export function renderWithProviders(
//   ui,
//   {
//     // preloadedState = {},
//     // Automatically create a store instance if no store was passed in
//     store = reduxStore,
//     ...renderOptions
//   } = {},
// ) {
//   function Wrapper({ children }) {
//     return <Provider store={store}>{children}</Provider>;
//   }

//   // Return an object with the store and all of RTL's query functions
//   return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
// }

renderWithProvidersOnly.propTypes = {
  children: PropTypes.node.isRequired,
};
