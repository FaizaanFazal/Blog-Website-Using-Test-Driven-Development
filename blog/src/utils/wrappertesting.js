import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import PropTypes from 'prop-types';
import reduxStore from '../Redux/store';
import '@testing-library/jest-dom';

export * from '@testing-library/react';

export function renderWithProviders(
  ui,
  {
    // preloadedState = {},
    // Automatically create a store instance if no store was passed in
    store = reduxStore,
    ...renderOptions
  } = {},
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  // Return an object with the store and all of RTL's query functions
  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}

renderWithProviders.propTypes = {
  children: PropTypes.node.isRequired,
};
