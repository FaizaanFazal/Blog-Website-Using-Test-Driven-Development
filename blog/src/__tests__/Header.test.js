import { fireEvent, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Header from '../components/Header/Header';
import { renderWithProviders } from '../utils/wrappertesting';

describe('Unit tests for Header', () => {
  it('Snapshot test for rendering navbar in header', () => {
    const { asFragment } = renderWithProviders(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('Dom test for-> renders the navbar', () => {
    renderWithProviders(<Header />);
    const navbarElement = screen.getByTestId('header');
    expect(navbarElement).toBeInTheDocument();
  });

  it('Updates input value on change in search bar', () => {
    renderWithProviders(<Header />);
    const inputSearch = screen.getByTestId('inputSearch');
    fireEvent.change(inputSearch, { target: { value: 'test' } });
    expect(inputSearch.value).toBe('test');
  });

  it('Number of tags should be 3', () => {
    renderWithProviders(<Header />);
    const tagElements = screen.getAllByTestId('tag');
    expect(tagElements.length).toBe(3);
  });

  it('Checks if button is disabled if search bar is empty', () => {
    renderWithProviders(<Header />);
    const searchbtn = screen.getByText('Search');
    const inputSearch = screen.getByTestId('inputSearch');
    fireEvent.change(inputSearch, { target: { value: '' } });
    expect(searchbtn).toBeDisabled();
  });
});
