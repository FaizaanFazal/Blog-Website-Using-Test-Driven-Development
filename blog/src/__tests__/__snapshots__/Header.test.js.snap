// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for Header Snapshot test for rendering navbar in header 1`] = `
<DocumentFragment>
  <header
    class="header flex flex-col"
  >
    <nav
      class="navbar flex items-center "
      role="navigation"
    >
      <div
        class="container navbar-container w-full flex justify-between"
      >
        <div
          class="brand-and-toggler flex items-center justify-between"
        >
          <a
            class="nav-brand"
            data-testid="logo"
            href="/"
          >
            Blogs 
            <span
              class="nav-brand-dot"
            />
          </a>
          <button
            class="nav-show-btn text-white"
            type="button"
          >
            <i
              class="bi bi-list"
            />
          </button>
        </div>
        <div
          class="nb-menu-wrapper flex items-center "
        >
          <button
            class="nav-hide-btn"
            type="button"
          >
            <i
              class="bi bi-x-square"
            />
          </button>
          <div>
            <ul
              class="nav-menu flex items-center"
            >
              <li
                class="nav-item"
                data-testid="listitem"
              >
                <a
                  class="nav-link"
                  href="/"
                  role="link"
                >
                  Home
                </a>
              </li>
              <li
                class="nav-item"
                data-testid="listitem"
              >
                <a
                  class="nav-link"
                  href="/about"
                  role="link"
                >
                  About
                </a>
              </li>
              <li
                class="nav-item"
                data-testid="listitem"
              >
                <a
                  class="nav-link"
                  href="/contacts"
                  role="link"
                >
                  Contact
                </a>
              </li>
              <div
                class="nav-btns"
              >
                <a
                  class="nav-btn btn"
                  data-testid="listitem"
                  href="/login"
                >
                  Login
                </a>
              </div>
              <div
                class="nav-btns"
              >
                <a
                  class="nav-btn btn"
                  data-testid="listitem"
                  href="/createpost"
                >
                  Post
                </a>
              </div>
            </ul>
          </div>
        </div>
      </div>
    </nav>
    <div
      class="container flex-1 w-full flex items-center justify-center"
      data-testid="header"
    >
      <div
        class="hd-content flex flex-col justify-center items-center text-center"
      >
        <h1
          class="hd-title text-white"
        >
          Our Newsroom
        </h1>
        <div
          class="hd-search"
        >
          <form>
            <div
              class="form-group flex items-stretch bg-white"
            >
              <span
                class="form-gp-icon items-center justify-center"
              >
                <i
                  class="bi bi-search"
                />
              </span>
              <input
                class="form-gp-input"
                data-testid="inputSearch"
                placeholder="Search article"
                type="text"
                value=""
              />
              <button
                class="form-gp-btn text-white bg-primary btn"
                disabled=""
                type="button"
              >
                Search
              </button>
            </div>
          </form>
        </div>
        <div
          class="hd-tags text-white flex items-center"
        >
          <h3
            class="hd-tags-title nowrap text-base"
          >
            Popular Tags:
          </h3>
          <ul
            class="tags-list flex flex-wrap items-center"
          >
            <li
              class="tags-item text-sm"
              data-testid="tag"
            >
              Design
            </li>
            <li
              class="tags-item text-sm"
              data-testid="tag"
            >
              User Experience
            </li>
            <li
              class="tags-item text-sm"
              data-testid="tag"
            >
              User Interfaces
            </li>
          </ul>
        </div>
      </div>
    </div>
  </header>
</DocumentFragment>
`;
